dashboard/app/routes/optimization.fine-tuning/ModelSelector.tsx

```diff
@@ -0,0 +1,93 @@
+import { Control } from "react-hook-form";
+import { FormField, FormItem, FormLabel } from "~/components/ui/form";
+import {
+  Select,
+  SelectContent,
+  SelectItem,
+  SelectTrigger,
+  SelectValue,
+} from "~/components/ui/select";
+import { SFTFormValues } from "./types";
+import { ModelOption } from "./model_options";
+
+type ModelSelectorProps = {
+  control: Control<SFTFormValues>;
+  models: ModelOption[];
+};
+
+function formatProvider(provider: string): { name: string; className: string } {
+  switch (provider) {
+    case "openai":
+      return {
+        name: "OpenAI",
+        className:
+          "bg-blue-100 text-blue-800 dark:bg-blue-900 dark:text-blue-300",
+      };
+    case "anthropic":
+      return {
+        name: "Anthropic",
+        className:
+          "bg-green-100 text-green-800 dark:bg-green-900 dark:text-green-300",
+      };
+    case "mistral":
+      return {
+        name: "Mistral",
+        className:
+          "bg-purple-100 text-purple-800 dark:bg-purple-900 dark:text-purple-300",
+      };
+    default:
+      return {
+        name: provider,
+        className:
+          "bg-orange-100 text-orange-800 dark:bg-orange-900 dark:text-orange-300",
+      };
```

Comment:
As is, this is bad. You forgot fireworks for example. There should be some test or typing or something that ensures this covers all the providers we need. TS types can likely solve it (`provider` with a custom type).